apps/src/code-studio/components/ModelManagerDialog.jsx

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import $ from 'jquery';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import Button from '@cdo/apps/templates/Button';
+
+export default class ModelManagerDialog extends React.Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    autogenerateML: PropTypes.func
+  };
+
+  state = {
+    models: [],
+    isImportPending: false
+  };
+
+  componentDidUpdate() {
+    this.getModelList();
```

Comment:
When running a regular AppLab app, we keep on hitting `/api/v1/ml_models/names`.  This component seems to be updating more than expected.

Reply:
`componentDidMount()` instead of `componentDidUpdate()` 🤦‍♀️

Reply:
That seems to fire once, but due to the lifetime management of these dialogs (they live forever, just showing and hiding), it doesn't refresh the model list each time the dialog is opened.  I have implemented what seems to be a solution in https://github.com/code-dot-org/code-dot-org/pull/39041.